sdk/lib/_internal/js_dev_runtime/private/js_string.dart

```diff
@@ -83,6 +83,104 @@ final class JSString extends Interceptor
     return stringReplaceAllFuncUnchecked(this, from, onMatch, onNonMatch);
   }
 
+  @notNull
+  Iterable<T> splitMap<T>(
+    Pattern pattern, {
+    T Function(Match match)? onMatch,
+    T Function(String nonMatch)? onNonMatch,
+  }) {
+    return stringSplitMapUnchecked(this, pattern, onMatch, onNonMatch);
+  }
+
+  Iterable<T> stringSplitMapUnchecked<T>(String receiver, Pattern pattern,
```

Comment:
(Just style comments, not sure this is the design we'll do.)
Should be private. All helper functions should.